### PR TITLE
Pan: add support for relative prefix statement

### PR DESCRIPTION
**Pull request by stdweird**
Support relative prefix, relative to previous defined absolute prefix.

E.g.
```pan
prefix "/a/b";
prefix "c"
"d" = 1;
```
is equal to 
```pan
prefix "/a/b";
"c/d" = 1;
```

### CODE REVIEW

**Comment by jrha** [comment]
Interesting! The behaviour may be a little bit surprising to some users, but LGTM.

**Comment by stdweird** [comment]
@jrha what could be surprising? that it is always relative to the absolute prefix, it's `prefix`,  not `cd` 😄

**Comment by jrha** [comment]
Yes that, and when someone does it by accident :wink:

File: panc-docs/source/pan-book/pan-book.rst [hunk 1]
```diff
@@ -815,6 +815,29 @@ included templates. Although multiple ``prefix`` statements can be used
 in a template, it is best practice to use only one near the beginning of
 the template.
 
+The ``prefix`` statement itself also supports a relative path. In that case,
+a previously defined ``prefix`` statement with an absolute path is required,
+and the relative prefix is relative to the (last) absolute prefix.
+This is very useful when dealing with very large paths and a complex
+schema (e.g. ``ncm-metaconfig`` configuration,
```
Comment (jouvin) [inline panc-docs/source/pan-book/pan-book.rst]: I would not reference configuration modules in the panc documentation. May be just done an example of a long path.

Comment (stdweird) [reply panc-docs/source/pan-book/pan-book.rst]: hmmm, i'd rather drop the example in that case.

**Comment by stdweird** [comment]
@jouvin i removed the metaconfig example

**Comment by stdweird** [comment]
@jrha @ned21 @jouvin any ETA on a release? we are writing and cleaning up bunch of templates, we could really use this (and some other features). if noone cares to review it, it's probably ok 😞

File: panc/src/main/java/org/quattor/pan/parser/PanParserAstUtils.java [hunk 2]
```diff
@@ -285,8 +285,9 @@ static public Template convertAstToTemplate(File file, ASTTemplate ast) throws S
         // Create a list containing all of the statements.
         LinkedList<Statement> statements = new LinkedList<Statement>();
 
-        // By default the prefix for paths is empty.
-        Path prefix = null;
+        // Prefixes list: first element is the prefix, 2nd element the absolute prefix
```
Comment (ned21) [inline panc/src/main/java/org/quattor/pan/parser/PanParserAstUtils.java]: I completely mis-read this at first. Can you add the following to make it more obvious how this works?

2nd element the most recent absolute prefix, to which any new relative prefix is prepended.

**Comment by stdweird** [comment]
@ned21 updated

**Comment by ned21** [comment]
I'm unconvinced by the CodeClimate test.. but then I'm not a Java programmer so I don't know whether it's valid or not! :(

**Pull request merged**